osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs

```diff
@@ -188,9 +188,9 @@ protected override bool OnKeyDown(KeyDownEvent e)
         protected override void UpdateState(ValueChangedEvent<Visibility> state)
         {
             base.UpdateState(state);
-
-            if (state.NewValue == Visibility.Hidden)
-                footerButton.IsActive.Value = false;
+            // intentionally scheduling to let the button have a chance whether the popover will hide from clicking the button or clicking outside
```

Comment:
This schedule might be okay, but this comment isn't. I read it five times and still can't understand what it's trying to say.

Reply:
Hmm, it does seem like I've condensed too much context inside. I'll rewrite it.

Reply:
How does this read?
```diff
diff --git a/osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs b/osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs
index 648f536bb1..ae928965a3 100644
--- a/osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs
+++ b/osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs
@@ -188,8 +188,9 @@ protected override bool OnKeyDown(KeyDownEvent e)
         protected override void UpdateState(ValueChangedEvent<Visibility> state)
         {
             base.UpdateState(state);
-            // intentionally scheduling to let the button have a chance whether the popover will hide from clicking the button or clicking outside
-            // see the "hidingFromClick" field in FooterButtonOptionsV2.
+
+            // a user might be clicking on the footer button itself to close this popover, but the popover responds to the "mouse down" first before the footer button does, due to its order in the input queue.
+            // to give the footer button a chance to know it's being clicked to close the popover (as it relies on overlay state), schedule updating the overlay state to the next frame.
             Schedule(() => footerButton.OverlayState.Value = state.NewValue);
         }
     }
```

Reply:
I dunno. Based on Joehuu finding at least one other case of this, maybe we should just leave it for now? I don't see it harming anyone (it's not an "annoying" interaction to me)? Doesn't feel important to fix it in this one location any more than others.

Here's another for good measure:


https://github.com/ppy/osu/assets/191335/63d90ce0-4f0a-4484-897d-fffa2981dfc0


I'd say revert and have this PR just fix the visuals.

Reply:
Unfortunate, but no problem.